### ISSUE TITLE
Indent/dedent with tab/shift-tab

### ISSUE THREAD

**Issue by dela3499** [twosigma/beakerx#3708]
I (mistakenly?) expect shift-tab to dedent, rather than bring up documentation in Python. Perhaps behavior of shift-tab should be sensitive to context (dedent selections, show documentation if nothing is selected). CTRL + Shift + F is a bit awkward. (Is there an option to customize?)

By the way, I didn't know that documentation was available, so I like the feature! :)

**Comment by dela3499** [comment]
Hmm. Now that I look more closely, it seems the behavior of Shift + Tab is already pretty complex. Sometimes it does dedent, depending on what you've selected. Sometimes is does nothing. Sometimes it brings up documentation.

Cursor in word (no selection) = documentation
Cursor left of word (no selection) = moves cursor to start of word
Select entire line of text = sometimes dedents

I guess I won't try all the test cases just yet, but it seems behavior depends where the cursor is placed, what is selected, and what code is around the selection. Hmm. Maybe this behavior can be simplified. 

---

Though just a tangent, it might also be nice for the documentation go away once you click outside the popup. Also, would be nice to have popup close after hitting escape only once, rather than twice.
